### ISSUE TITLE
waypoint sharing failing on server

### ISSUE THREAD

**Issue by facetorched** [1whohears/JourneyMapQOL_1.7.10#2]
I am able to share waypoints (with myself) in singleplayer but if I create a server (with only this mod and journeymap) and join with a client with the same mods. I attempt the following:

- create a waypoint named test
- `/waypoint share test`

I get the following error and the waypoint isn't shared:

```
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: java.lang.NullPointerException
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at onewhohears.minecraft.jmapi.command.WaypointCommand.func_71515_b(WaypointCommand.java:139)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at net.minecraftforge.client.ClientCommandHandler.func_71556_a(ClientCommandHandler.java:72)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at net.minecraft.client.gui.GuiChat.func_146403_a(GuiChat.java:141)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at net.minecraft.client.gui.GuiChat.func_73869_a(GuiChat.java:131)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at net.minecraft.client.gui.GuiScreen.func_146282_l(GuiScreen.java:319)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:276)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1640)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:973)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:898)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at net.minecraft.client.main.Main.main(SourceFile:148)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at java.lang.reflect.Method.invoke(Method.java:497)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
[03/09/2022 14:38:13 PM] [Client thread/INFO] [java.lang.Throwable$WrappedPrintStream:println:748]: 	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
```

**Comment by 1whohears** [comment]
Sad. The reason why they stopped working for servers on their own machine is because `WaypointCommand` calls `ApiWaypointManager.instance` to do client side functions. But that object is only generated in a server startup event which fires on the machine the server is in. So if the server is not on the same machine as the client `ApiWaypointManager.instance = null`. In previous versions I didn't use that api class to do client commands. But at some point I did for simplicity sake and only tested those changes with the server and the client on the same machine. I didn't realize that the client was using the server's instance of the api class and assumed the client was making it's own. 
Thank you so much for the report. I took a break from this project a while ago and started playing newer versions of the game lol. I would have never noticed this.
